scripts/run_registry.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash  
+
+set -e
+
+SCRIPT=$(realpath -s "${0}")
+SCRIPTS_PATH=$(dirname "${SCRIPT}")
+TESTS_PATH=$(realpath -s "${SCRIPTS_PATH}/../tests")
+
+: "${ELMNTL_PREFIX:=}"
+: "${ELMNTL_REGCONF:=${TESTS_PATH}/${ELMNTL_PREFIX}test-registry.yaml}"
+: "${ELMNTL_IPADDR=$(ip -4 addr show eth0 | grep -oP '(?<=inet\s)\d+(\.\d+){3}')}"
+: "${ELMNTL_REGPORT=5000}"
+
+function _abort {
+    echo "$@" && exit 1
+}
+
+function start {
+  if [ "$( docker container inspect -f '{{.State.Status}}' elmntl_registry )" = "running" ]; then
+    return
+  fi
+
+  docker run --rm -d \
+    --name elmntl_registry \
+    -p "${ELMNTL_REGPORT}:${ELMNTL_REGPORT}" \
+    registry:2
+}
+
+function stop {
+  docker stop elmntl_registry || true
+  ([ -f "${ELMNTL_REGCONF}" ] && rm "${ELMNTL_REGCONF}") || true
+}
+
+function push {
+  local reg_img
+
+  for img in "$@"; do
+    reg_img="${ELMNTL_IPADDR}.sslip.io:${ELMNTL_REGPORT}/${img}"
+    echo "Pushing '${img}' to '${reg_img}'"
+
+    # Ugly hack around podman to circumvent the need of adding insecure registries
+    # at /etc/docker/daemon.json when using the docker client 
+    docker save "${img}" | podman load
+    tag=${img##*:}
+    imgID=$(podman images -n | grep "${tag}" | awk '{print $3}' | head -n1)
+    podman tag "${imgID}" "${reg_img}"
+    podman push --tls-verify=false "${reg_img}"
+    podman rmi -f "${imgID}"
```

Comment:
The ugliness of all this is two fold:

* We are using `podman push` because docker does not have the equivalent of `--tls-verify=false` and the daemon needs to be configured for that. I want `make test-upgrade` to be executable locally without having to mess with docker daemon configuration.
* GH Runners use podman v3.4.4 (current podman is at v5.1) and this version does not properly keep image reference on `podman load`, so that there is this `podman images` to find the actual image ID, because we can't relay on the image being loaded with the former reference 😞

Reply:
I think GH Actions still uses `ubuntu-22.04` by default, but it should be safe to upgrade to `ubuntu-24.04`instead, even if it's beta. That should contain podman 4.9.3

Reply:
I had to roolback to 22.04 as KVM support fails in 24.04, I think it would be better to wait until manage to run KVM on 24.04, out of the box did not work.